src/ServicePulse.Host/vue/src/components/failedmessages/FlowDiagram.vue

```diff
@@ -61,292 +58,176 @@ function mapMessage(message) {
-    messages[i].children = []; // initialize the children
-  }
+  elements.value = [...constructNodes(mappedMessages), ...constructEdges(mappedMessages)];
+  //TODO: if doing fitView on next Vue onUpdated() then it doesn't appear the elements
+  // are actually drawn yet. See if we can determine a better event to use this on rather than relying on setTimeout
+  setTimeout(() => fitView(), 50);
+});
 
-  for (i = 0; i < messages.length; i += 1) {
-    node = messages[i];
-    if (node.parentId && map[node.parentId]) {
-      // if you have dangling branches check that map[node.parentId] exists
-      messages[map[node.parentId]].children.push(node);
-    } else {
-      roots.push(node);
-    }
-  }
+onPaneReady(({ fitView }) => {
+  fitView();
+});
 
-  return roots;
+function typeIcon(type) {
+  //TODO can this be done without the type magic strings?
+  switch (type) {
+    case "Timeout message":
+      return "pa-flow-timeout";
+    case "Event message":
+      return "pa-flow-event";
+    default:
+      return "pa-flow-command";
+  }
 }
+</script>
 
-function drawTree(treeData) {
-  // Assigns parent, children, height, depth
-  root = hierarchy(treeData, function (d) {
-    return d.children;
-  });
-
-  // append the svg object to the body of the page
-  // appends a 'group' element to 'svg'
-  // moves the 'group' element to the top left margin
-  parentSvg = select("#tree-container").append("svg").attr("viewBox", "-1000 -10 2000 2000");
-
-  svg = parentSvg.append("g").attr("transform", "scale(.7,.7)");
-
-  var zoomElement = zoom()
-    .scaleExtent([1 / 2, 8])
-    .on("zoom", (event) => {
-      svg.attr("transform", event.transform);
-    });
-
-  svg
-    .append("defs")
-    .append("marker")
-    .attr("id", "end-arrow")
-    .attr("viewBox", "0 -5 10 10")
-    .attr("refX", 0)
-    .attr("refY", 0)
-    .attr("markerWidth", 6)
-    .attr("markerHeight", 6)
-    .attr("orient", "auto")
-    .attr("class", "arrow")
-    .append("path")
-    .attr("d", "M10,-5L0,0L10,5");
-
-  // declares a tree layout and assigns the size
-  treemap = tree().nodeSize([rectNode.width + 20, rectNode.height]);
-
-  root.x0 = width / 2;
-  root.y0 = 0;
+<template>
+  <div id="tree-container">
+    <VueFlow v-model="elements" :min-zoom="0.1">
+      <template #node-message="nodeProps">
+        <div class="node" :class="[nodeProps.data.type, nodeProps.data.isError && 'error', nodeProps.data.id === props.messageId && 'current-message']">
+          <div class="node-text wordwrap">
+            <i v-if="nodeProps.data.isError" class="fa pa-flow-failed" />
+            <i class="fa" :class="typeIcon(nodeProps.data.type)" :title="nodeProps.data.type" />
+            <div class="lead righ-side-ellipsis" :title="nodeProps.data.nodeName">
+              <strong>
+                <a v-if="nodeProps.data.isError" @click="() => $router.push({ path: `/failed-messages/message/${nodeProps.data.id}` })" href="javascript:void(0)">{{ nodeProps.data.nodeName }}</a>
```

Comment:
So do we still need the `href` here?

Reply:
as discussed in [this issue](https://github.com/Particular/ServicePulse/pull/1676), all the anchor's in the solution should be addressed together